lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -353,13 +353,18 @@ def generate_test_result_tree(tmp_path: Path, dir_name: str) -> Path:
                         subdir11/
                         subdir12/
                             f121_sym -> /tmp/<dir_name>/subdir1/subdir15
+                            f122_sym -> ./bad_subdir/nonexixtent_file.txt
                         subdir13/
                             f131_sym -> /etc/passwd
                         subdir14/
                             subdir141/
                                 f1411.txt
                                 f1412_sym -> /tmp/<dir_name>/subdir1/f11.txt
+                                f1413_sym -> ./subdir141
+                                f1414_sym -> ./subdir141/f1411.txt
+                                f1415_sym -> ./subdir141/f1412_sym
```

Comment:
I see what happened: these symlink files landed in the wrong subdirectory relative to [what I suggested](https://github.com/distributed-system-analysis/pbench/pull/3396/files/169b2266a462790db1c66d810a426992fae16c22#r1198469257), which is why their references are bad.  That is, instead of changing what they point _to_, you could change where they point **from**.